iree/turbine/kernel/boo/ops/graph.py

```diff
@@ -0,0 +1,186 @@
+    schema = "("
+    schema += ", ".join([f"Tensor {inp}" for inp in inputs])
+    schema += ") -> ("
+    schema += ", ".join([ret_ty for _ in outputs])
+    schema += ")"
+    return schema
+
+
+def get_mlir_module(gm: GraphModule) -> Any:
+    """Generates torch-mlir IR from a graph module."""
+    sample_args = tuple(
+        [n.meta.get("val") for n in gm.graph.nodes if n.op == "placeholder"]
+    )
+    if any([arg is None for arg in sample_args]):
+        raise ValueError("Provided gm does not have sufficient metadata.")
+    gm = turbine_cpu_pass_pipeline(gm, sample_args)
+    imp = FxImporter()
+    imp.import_graph_module(gm)
+    exp_ops = ExpandCustomOpsPass(imp.module_op)
+    exp_ops.run()
+    return imp.module_op
+
+
+def get_custom_graph_op(gm: GraphModule) -> Callable[[Any], Any]:
+    """Converts a graph module into a custom operator."""
+    hash = sha1(str(gm).encode(), usedforsecurity=False).hexdigest()
+    op_name = f"fused_op_{hash}"
+
+    if not hasattr(torch.ops.boo, op_name):
+        define_custom_graph_op(gm, op_name)
+
+    return get_library_op(op_name)
+
+
+def define_custom_graph_op(gm: GraphModule, op_name: str):
+    """Defines a custom op from the graph module with given op_name in the boo library."""
+    inputs, outputs = get_io_from_gm(gm)
+    # TODO: handle this better
+    is_none_output = maybe_trim_none_outputs(gm)
+    has_a_none_output = any(is_none_output)
+    schema = get_schema(inputs, outputs)
+    define_schema(op_name, schema)
+
+    @register_impl(op_name)
+    def _(*args):
+        spec_name = get_arg_spec_name(op_name, *args)
+        l = get_launchable(gm, arg_factory=args, func_name=spec_name)
+        results = l(*[arg.data for arg in args])
+        if not has_a_none_output:
+            return results
+        # Handle this better.
+        if isinstance(results, torch.Tensor):
+            results = [results]
+        all_results = []
+        i = 0
+        for is_none in is_none_output:
+            if is_none:
+                all_results.append(None)
+            else:
+                all_results.append(results[i])
+                i += 1
+        # It is probably safe to assume all_results has more than one value.
+        return tuple(all_results) if i > 1 else all_results[0]
+
+    @register_meta(op_name)
+    def _meta(*args):
+        outputs = gm.forward(*args)
+        if len(outputs) == 1:
+            return outputs[0]
+        return outputs
+
+
+def maybe_trim_none_outputs(gm: GraphModule) -> Sequence[bool]:
+    """Removes None outputs from graph. The ith return indicates whether output[i] was None."""
+    none_output = []
+    for n in gm.graph.nodes:
+        if n.op == "output":
+            trunc_returns = []
+            for ret in n.args[0]:
+                if ret is None:
+                    none_output.append(True)
+                    continue
+                trunc_returns.append(ret)
+                none_output.append(False)
```

Comment:
Something along these lines

```python
is_none = [ret is None for ret in n.args[0]]
trunc_returns = [ret for ret in in.args[0] if ret is not None]
```

will be more compact and slightly more efficient.